docs/specifications/tests/Zone-TP/zone11.md

```diff
@@ -0,0 +1,169 @@
+ 
+This Test Case will check that:
+ - the MNAME field contains the master name server.
+ - this master name server is authoritative to the zone.
+ - the SOA SERIAL of the MNAME is identical or higher than the ones found from the name servers in the delegation.
+ - the master name server is listed as part of the delegation of the zone.
+
+Note that the SOA SERIAL comparison must follow [RFC1982].
+
+## Scope
+
+It is assumed that *Child Zone* has been tested and reported by [Basic04]. This
+test case will just ignore non-responsive name servers or name servers not
+giving a correct DNS response for an authoritative name server.
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Summary
+
+Message Tag                   | Level   | Arguments            | Message ID for message tag
+:---------------------------- |:--------|:---------------------|:---------------------------------------------------------------------------------------
+Z11_MULTIPLE_MNAME            | WARNING | ns_ip_list           | Distinct MNAMEs found, but only one is expected. Returned from name servers "{ns_ip_list}".
+Z11_MNAME_NOT_AUTHORITATIVE   | WARNING | ns                   | MNAME name server points to a non authoritative name server ("{ns}").
+Z11_MNAME_NO_RESPONSE         | NOTICE  | ns                   | MNAME name server points to a non responsive name server ("{ns}").
+Z11_MNAME_NOT_IN_DELEGATION   | INFO    | nsname               | MNAME name server {nsname} is not listed as NS record for the zone.
+Z11_MNAME_NOT_MASTER          | WARNING | ns_ip_list           | MNAME name server doesn't have the highest SERIAL. Returned from name servers "{ns_ip_list}".
+Z11_MNAME_IS_MASTER           | DEBUG   | ns_ip_list           | MNAME name server does appear to be master. Returned from name servers "{ns_ip_list}".
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+## Test procedure
+
+In this section and unless otherwise specified below, the terms "[DNS Query]"
+follow the specification for DNS queries as specified in [DNS Query and Response Defaults].
+The handling of the DNS responses on the DNS queries follow, unless otherwise specified below,
+what is specified for [DNS Response] in the same specification.
+
+1. Create a [DNS Query] with query type SOA and query name *Child Zone* ("SOA Query")
+
+2. Obtain the set of name server IP addresses using [Method4] and [Method5] ("Name Server IP").
+
+3. Create the following empty sets:
+   1. Name server IP address, MNAME, SERIAL ("MNAME Nameserver")

```

Comment:
This should be `1. Name server name ("MNAME Nameserver")` to match the population in 4.3 below.

Reply:
This set is also used in step 6.1.2 to store the SERIAL field of the MNAME name server. And the IP address is an implicit requirement, from a coding perspective, for sending the query (step 6) to the MNAME name server.

Reply:
On second thoughts I will indeed remove the IP address from this set since it is covered by a DNS lookup from step 6.